### PR TITLE
[#155922718] user can see assets allocated to them via API

### PR DESCRIPTION
**Pull request by iankigen**
#### What does this PR do?

Allow users to view assets allocated to them via API

#### Description of Task to be completed?

- add test for functionality
- add item serializer
- add item viewset
- register item url

#### How should this be manually tested?

`python manage.py test`
or 
endpoint `/api-auth/login` to login user
endpoint `/api/v1/items` to view assets

#### Any background context you want to provide?

Included `api-auth` route to allow user login
#### What are the relevant pivotal tracker stories?

[#155922718](https://www.pivotaltracker.com/n/projects/2146417/stories/155922718)

#### Screenshots 
non_auth
test_user
test_user_1

[delivers #155922718

### CODE REVIEW

**Comment by iankigen** [comment]
@egichuri  @alexkiura  @nelmandela  @HawiCaesar Kindly review

File: api/tests/test_item_api.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        itemmodel.save()
+        item = Item(
+            item_code="IC001",
+            serial_number="SN001",
+            assigned_to=self.user,
+            model_number=itemmodel
+        )
+        item.save()
+        self.item = item
+        self.items_url = "/api/v1/items/"
```
Comment (egichuri) [inline api/tests/test_item_api.py]: i think we should start using `reverse`

Comment (iankigen) [reply api/tests/test_item_api.py]: I agree. On it

**Comment by iankigen** [comment]
@egichuri @alexkiura @nelmandela Kindly review

File: api/views.py [hunk 2]
```diff
@@ -12,3 +13,14 @@ class UserViewSet(ModelViewSet):
     queryset = User.objects.all()
     permission_classes = (IsAuthenticated, IsAdminUser)
     http_method_names = ['get', 'post']
+
+
+class ItemViewSet(ModelViewSet):
+    serializer_class = ItemSerializer
+    permission_classes = [IsAuthenticated, ]
+    http_method_names = ['get']
```
Comment (HawiCaesar) [inline api/views.py]: Does the `delete`, `patch` and `put` need to be here as you have endpoints for those http verbs ?

Comment (iankigen) [reply api/views.py]: @HawiCaesar The story limits only `list` for the normal user. Admin user may be allowed to `delete`, `patch` and `put`

**Pull request merged**